pkg/contract-manager/cr_services.go

```diff
@@ -0,0 +1,151 @@
+	flavours, err := getAllFlavours(cl)
+	if err != nil {
+		klog.Errorf("Error getting all the Flavour CRs: %s", err)
+		return nil, err
+	}
+
+	// Get the Flavour that matches the flavourID
+	for _, flavourCR := range flavours {
+		if flavourCR.Name == flavourID {
+			// Return from the Flavour CR the Flavour struct
+			flavour := parseCRToFlavour(flavourCR)
+			return flavour, nil
+		}
+	}
+
+	return nil, nil
+}
+
+// getFlavourByIDCR returns the entire Flavour CR (not only spec) in the cluster that matches the flavourID
+func getFlavourByIDCR(flavourID string, cl client.Client) (*nodecorev1alpha1.Flavour, error) {
+	flavoursSpec, err := getAllFlavours(cl)
+	if err != nil {
+		klog.Errorf("Error getting all the Flavour CRs: %s", err)
+		return nil, err
+	}
+
+	// Get the Flavour that matches the flavourID
+	for _, flavour := range flavoursSpec {
+		if flavour.Name == flavourID {
+			return &flavour, nil
+		}
+	}
+
+	return nil, nil
+}
+
+// filterFlavoursBySelector returns the Flavour CRs in the cluster that match the selector
+func filterFlavoursBySelector(flavours []localResourceManager.Flavour, selector Selector) ([]localResourceManager.Flavour, error) {
+	var flavoursSelected []localResourceManager.Flavour
+
+	// Get the Flavours that match the selector
+	for _, f := range flavours {
+		if f.Type == selector.FlavourType {
+			if f.Characteristics.CPU >= selector.CPU && f.Characteristics.RAM >= selector.RAM {
+				flavoursSelected = append(flavoursSelected, f)
+			}
+		}
+	}
+
+	return flavoursSelected, nil
+}
+
+// createContractCustomResourceSeller creates a Contract CR
+func forgeContract(ctx context.Context, flavour nodecorev1alpha1.Flavour, buyerID string) *reservationv1alpha1.Contract {
+	contract := &reservationv1alpha1.Contract{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "contract-" + flavour.Name,
+			/// TODO: diff namespace for different resource type
+			Namespace: discoverymanager.DEFAULT_NAMESPACE,
+		},
+		Spec: reservationv1alpha1.ContractSpec{
+			FlavourID: flavour.Name,
+			Buyer: nodecorev1alpha1.NodeIdentity{
+				NodeID: buyerID,
+			},
+			Seller: flavour.Spec.Owner,
+		},
+		Status: reservationv1alpha1.ContractStatus{
+			Phase: nodecorev1alpha1.PhaseStatus{
+				Phase: nodecorev1alpha1.PhaseRunning,
+			},
+		},
+	}
+
+	return contract
+}
+
+// parseFlavourSpecToFlavour converts a FlavourSpec to a Flavour struct
+func parseCRToFlavour(flavourCR nodecorev1alpha1.Flavour) *localResourceManager.Flavour {
+
+	// It is converted in int since REAR for now only supports int
+	cpuInt, _ := flavourCR.Spec.Characteristics.Cpu.AsInt64()
+	ramInt, _ := flavourCR.Spec.Characteristics.Memory.AsInt64()
+
+	flavour := &localResourceManager.Flavour{
+		FlavourID:  flavourCR.Name,
+		ProviderID: flavourCR.Spec.ProviderID,
+		Type:       string(flavourCR.Spec.Type),
+		Characteristics: localResourceManager.Characteristics{
+			CPU: int(cpuInt),
+			RAM: int(ramInt),
+		},
```

Comment:
few characteristics! what happen if I have other fields? as i said a lot of time we need to foresee everywhere all the possible k8s resource characteristics (the standard ones).

Reply:
And now we have more characteristics, hope these will be fine